### ISSUE TITLE
`quarto publish gh-pages`: use remote source other than `origin`

### ISSUE THREAD

**Issue by cderv** [quarto-dev/quarto-cli#5997]
### Discussed in https://github.com/quarto-dev/quarto-cli/discussions/5993

<div type='discussions-op-text'>

<sup>Originally posted by **rgaiacs** June 21, 2023</sup>
### Description

I have a Git remote source named `quarto`:

```
$ git remote -v
quarto  git@github.com:gesiscss/orc2.git (fetch)
quarto  git@github.com:gesiscss/orc2.git (push)
```

Executing `quarto publish gh-pages` produces the follow prompt

```
? Publish site to undefined using gh-pages? (Y/n) › 
```

Will be great if the prompt get information of other remote sources.</div>

**Comment by mcanouil** [comment]
FYI:

https://github.com/quarto-dev/quarto-cli/blob/41501dd86dee2103cb5d88954303feed32f1d3b0/src/core/github.ts#L40

**Comment by cderv** [comment]
Yes I agree with you @cscheid. This is initially why I consider this a bug. I thing it should be a better message and possibly error if another remote than origin is not supported.

**Comment by mcanouil** [comment]
To note, this was actually a duplicate of:
- #1249

**Comment by petrelharp** [comment]
Related: in [the documentation](https://quarto.org/docs/publishing/github-pages.html#publish-action) there is no mention of this requirement, so if for instance the remote is named `upstream` (also fairly common) then the instructions given at https://quarto.org/docs/publishing/github-pages.html#publish-action will fail, rather inscruitibly.

**Comment by mcanouil** [comment]
@petrelharp We are aware that the current documentation for publishing to GitHub pages is outdated, see:
- #6204

**Comment by amael-ls** [comment]
Hello,

I found this discussion when I was trying to do the same thing (with an origin called `origin-github`). Is there any plan about it? I was thinking about a new option, something like:

`quarto publish gh-pages --remote <NAME_OF_YOUR_ORIGIN>`

Many thanks, for the development of Quarto, when I think I used Latex for 15 years!

**Comment by mcanouil** [comment]
>  Is there any plan about it?

That's the purpose of an issue being opened. If something is not planned or considered, it would be closed.

Because of time constraint and likely because you don't really need `quarto publish gh-pages` to actually publish to GitHub Pages, it's not a short-term priority I believe.

**Comment by seanricardoford** [comment]
+1 for the --remote option. My internal remote "origin" is to gitlab and my external remote "publish" is to github (to use github-pages).

**Comment by mcanouil** [comment]
@seanricardoford use reactions on the original post to show interest rather than commenting and triggering notifications for all participants.
